index.src.html

```diff
@@ -546,25 +519,20 @@ <h4 id="get-types">
   1.  If |response| does not contain a <a>`Clear-Site-Data`</a> header, return
       an empty list.
 
-  2.  Let |types| be an empty list.
-
-  3.  Let |list| be the result of executing the algorithm defined in Section 4
-      of [[!HTTP-JFV]] on the value of |response|'s <a>`Clear-Site-Data`</a>
-      header. If that algorithm results in an error, return an empty list.
-
-  4.  For each |item| in |list|:
+  2.  Let |recognizedTypes| be an empty list.
 
-      1.  If |item| does not have a <a>`types`</a> member, skip to the next
-          |item|.
+  3.  If the value of the <a>`Clear-Site-Data`</a> header is a valid
+      <a grammar>clear-site-data-value</a>, let |types| be the list of all
+      occurences of <a grammar>type</a> in it. Otherwise, return an empty list.
 
-      2.  For each |type| in |item|'s <a>`types`</a> member's value:
+  4.  For each |type| in |types|:
 
-          1.  If |type| is <a>`cache`</a>, <a>`cookies`</a>, <a>`storage`</a>,
-              or <a>`executionContexts`</a>, append |type| to |types|.
+        1.  If |type| is <a>`cache`</a>, <a>`cookies`</a>, <a>`storage`</a>,
+            or <a>`executionContexts`</a>, append |type| to |recognizedTypes|.
 
-              Otherwise, skip to the next |type|.
+            Otherwise, skip to the next |type|.
 
-  5.  Return |types|.
+  5.  Return |recognizedTypes|.
```

Comment:
Done. Agree with returning failure in level 3, this is what Chromium's implementation actually does (we print an error to the console, we don't just ignore it).

Clarifying question: Is it OK to omit the old step #1 which checked if the header exists? Because if we run the current version of the algorithm on any random response, it will return failure. My interpretation of that would be that the browser outputs a console error every time it doesn't see a `Clear-Site-Data` header. Or is it assumed from the context that the algorithm is only executed for responses that do have the header?